File: app/src/main/java/com/example/jobdemo/dialog/BottomDialog.java
package com.example.jobdemo.dialog;

import android.os.Bundle;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;
import android.view.WindowManager;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatDialogFragment;

import com.example.jobdemo.R;

/**
 * @Author LYX
 * @Date 2022/10/11 15:30
 */
public abstract class BottomDialog extends AppCompatDialogFragment {


    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        //设置style
        setStyle(AppCompatDialogFragment.STYLE_NORMAL, R.style.BottomDialogStyle);
    }

    @Override
    public void onStart() {
        super.onStart();
        //设置 dialog 的宽高
        getDialog().getWindow().setLayout(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT);
        //设置 dialog 的背景为 null
        getDialog().getWindow().setBackgroundDrawableResource(android.R.color.transparent);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {

        //去除标题栏
        getDialog().requestWindowFeature(Window.FEATURE_NO_TITLE);
        Window window = getDialog().getWindow();
        WindowManager.LayoutParams lp = window.getAttributes();
        lp.gravity = Gravity.BOTTOM;
        lp.width = WindowManager.LayoutParams.MATCH_PARENT;
        window.setAttributes(lp);
        return createView(inflater, container);
    }


    /**
     * 重写此方法，设置布局文件
     *
     * @param inflater  布局
     * @param container 父控件
     * @return view
     */
    protected abstract View createView(LayoutInflater inflater, ViewGroup container);


}

File: app/src/main/java/com/example/jobdemo/kotlin_code/dialog/BottomDialog.kt
package com.example.jobdemo.kotlin_code.dialog

import android.app.Dialog
import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.jobdemo.R
import com.example.jobdemo.databinding.LayoutBottomDialogBinding
import com.example.jobdemo.kotlin_code.adapter.BottomDialogAdapter
import com.example.jobdemo.util.ToastUtils
import com.example.jobdemo.util.Utils
import com.google.android.material.bottomsheet.BottomSheetDialog
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import com.google.android.material.divider.MaterialDividerItemDecoration

/**

 * @Author LYX

 * @Date 2023/1/2 17:53

 */
class BottomDialog : BottomSheetDialogFragment() {
//    override fun onCreateView(
//        inflater: LayoutInflater,
//        container: ViewGroup?,
//        savedInstanceState: Bundle?
//    ): View {
//        val binding = LayoutBottomDialogBinding.inflate(inflater, container, false)
//        val defaultSentenceArray: Array<String> = if (Utils.getUserType(context)) {
//            resources.getStringArray(R.array.companyOftenSentence)
//        } else {
//            resources.getStringArray(R.array.normalOftenSentence)
//        }
//        val list: MutableList<String> = ArrayList()
//        list.addAll(defaultSentenceArray)
//        binding.rvSentenceList.layoutManager = LinearLayoutManager(context)
//        context?.let {
//            val decoration =
//                MaterialDividerItemDecoration(it, MaterialDividerItemDecoration.VERTICAL)
//            decoration.dividerColor = ContextCompat.getColor(it, R.color.lightGray)
//            binding.rvSentenceList.addItemDecoration(decoration)
//        }
//        binding.rvSentenceList.adapter = BottomDialogAdapter(list)
//        binding.tvAdd.setOnClickListener { ToastUtils.shortToast(context, "点击了添加") }
//        binding.tvRevamp.setOnClickListener { ToastUtils.shortToast(context, "点击了修改") }
//        return binding.root
//    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val binding = LayoutBottomDialogBinding.inflate(layoutInflater)
        val defaultSentenceArray: Array<String> = if (Utils.getUserType(context)) {
            resources.getStringArray(R.array.companyOftenSentence)
        } else {
            resources.getStringArray(R.array.pictureUrls)
        }
        val list: MutableList<String> = ArrayList()
        list.addAll(defaultSentenceArray)
        binding.rvSentenceList.layoutManager = LinearLayoutManager(context)
        context?.let {
            val decoration =
                MaterialDividerItemDecoration(it, MaterialDividerItemDecoration.VERTICAL)
            decoration.dividerColor = ContextCompat.getColor(it, R.color.lightGray)
            binding.rvSentenceList.addItemDecoration(decoration)
        }
        binding.rvSentenceList.adapter = BottomDialogAdapter(list)
        binding.tvAdd.setOnClickListener { ToastUtils.shortToast(context, "点击了添加") }
        binding.tvRevamp.setOnClickListener { ToastUtils.shortToast(context, "点击了修改") }


        val root: View = binding.root
        //这样创建dialog就这样传递style
        val dialog = BottomSheetDialog(requireContext(), R.style.TransBottomSheetDialogStyle)
        dialog.setContentView(root);
        //设置宽度
        val params: ViewGroup.LayoutParams = root.layoutParams
        params.height = (0.75 * resources.displayMetrics.heightPixels).toInt()
        root.layoutParams = params
        return dialog
    }

//    override fun getTheme(): Int {
//        return R.style.TransBottomSheetDialogStyle
//    }
}